### PR TITLE
Places Autocomplete (part 1)

### PR DESCRIPTION
**Pull request by ecgreb**
### Overview
Adds PlaceAutocompleteFragment widget.

### Proposed Changes
Adds autocomplete widget for performing geographic search.

### Depends on
Depends on https://github.com/mapzen/android/pull/234 to resolve OkHttp version conflicts in Mapzen Places module.

### TODO
The following items are outstanding for full autocomplete support and could be completed as part of this pull request or as follow-up issues:

#196

### CODE REVIEW

File: mapzen-places-api/src/main/java/com/mapzen/places/api/internal/ui/PlaceAutocompleteActivity.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    Pelias pelias = new Pelias();
+    pelias.setDebug(true);
+    pelias.setRequestHandler(new PeliasRequestHandler() {
+      @Override public Map<String, String> headersForRequest() {
+        return null;
+      }
+
+      @Override public Map<String, String> queryParamsForRequest() {
+        HashMap<String, String> params = new HashMap<>();
+        params.put("api_key",  "mapzen-ZTA7CTR");
```
Comment (sarahlensing) [inline mapzen-places-api/src/main/java/com/mapzen/places/api/internal/ui/PlaceAutocompleteActivity.java]: Did you mean to commit this?

File: mapzen-places-api/src/main/java/com/mapzen/places/api/internal/ui/PlaceAutocompleteActivity.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        return params;
+      }
+    });
+
+    pelias.setLocationProvider(new PeliasLocationProvider() {
+      @Override public double getLat() {
+        return 40.7443;
```
Comment (sarahlensing) [inline mapzen-places-api/src/main/java/com/mapzen/places/api/internal/ui/PlaceAutocompleteActivity.java]: I see your todo item around setting these values properly but for now maybe its better to have null island values?

File: mapzen-places-api/src/main/java/com/mapzen/places/api/internal/ui/PlaceAutocompleteActivity.java (hunk 1 above)
Comment (ecgreb) [inline mapzen-places-api/src/main/java/com/mapzen/places/api/internal/ui/PlaceAutocompleteActivity.java]: Oops will remove this was a debug key from a random demo but yes should be removed

File: mapzen-places-api/src/main/java/com/mapzen/places/api/internal/ui/PlaceAutocompleteActivity.java (hunk 2 above)
Comment (ecgreb) [inline mapzen-places-api/src/main/java/com/mapzen/places/api/internal/ui/PlaceAutocompleteActivity.java]: Hardcoding location to the office in NY tended to produce better results than null island

File: mapzen-places-api/src/main/java/com/mapzen/places/api/internal/ui/PlaceAutocompleteActivity.java (hunk 1 above)
Comment (ecgreb) [inline mapzen-places-api/src/main/java/com/mapzen/places/api/internal/ui/PlaceAutocompleteActivity.java]: Removed.

File: mapzen-places-api/src/main/java/com/mapzen/places/api/internal/ui/PlaceAutocompleteActivity.java (hunk 2 above)
Comment (ecgreb) [inline mapzen-places-api/src/main/java/com/mapzen/places/api/internal/ui/PlaceAutocompleteActivity.java]: Added TODO inline to replace this with real location integration

**Pull request merged**